### DISCUSSION TITLE
Deploy nlohmann-json package to platform-independent package repository

### DISCUSSION THREAD

**Discussion by mgovers** [nlohmann/json#4157] [Show and tell]
* We upload [an `nlohmann-json` package to PyPI](https://pypi.org/project/nlohmann-json/) as an independent repackaging of the [`nlohmann-json` package in HomeBrew](https://formulae.brew.sh/formula/nlohmann-json) our organization created in [this repo](https://github.com/PowerGridModel/nlohmann-json-pypi).
  * For the time being, we own that repackaging software
  * We are, however, very happy to transfer ownership of this software and the [`nlohmann-json` package in PyPI](https://pypi.org/project/nlohmann-json/) to the [nlohmann](https://github.com/nlohmann) organization
  * If you would like any help to find a more clean way to repackage or push to [PyPI](https://pypi.org/project/nlohmann-json/), we are very open to providing input.

**Issue**

Platform- and environment-independently downloading the latest release in automation software is flaky.

**Background**

In our workflow, we would like to build a Python wrapper around a C library. If possible, we would like to do so within the Python package builder without having to care about any C++ package manager. The user should be able to install the Wheel of the Python package while only depending on other Python packages.

In our ecosystem, we enable the above, by downloading the latest released version of the `nlohmann-json` package in automation and scripts via the GitHub API. However, the [GitHub API has a rate limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting) that is very regularly exceeded, resulting in `403` errors (see for example [this](https://github.com/PowerGridModel/power-grid-model/actions/runs/6249740895/job/16967137612#step:5:1) pipeline run). A workaround could be to use the direct HTTPS URL instead of the API, but that also has a rate limit, albeit a bit higher. It would be no more than a temporary fix.

**Proposed solution**

Push the `nlohmann-json` header-only package to multiple package managers for platform- and environment-independent usage.

**Describe alternatives you have considered**

* [`nlohmann-json` package in HomeBrew](https://formulae.brew.sh/formula/nlohmann-json) is not platform independent
* [`nlohmann_json` package in conda-forge](https://anaconda.org/conda-forge/nlohmann_json) may not be available everywhere by design (e.g.: when building Python bindings for non-conda environments)
* [`nlohmann-json` package in PyPI](https://pypi.org/project/nlohmann-json/) is an independent repackaging of the [`nlohmann-json` package in HomeBrew](https://formulae.brew.sh/formula/nlohmann-json) our organization created in [this repo](https://github.com/PowerGridModel/nlohmann-json-pypi). See also the top of this discussion.
  * For the time being, we own that repackaging software
  * We are, however, very happy to transfer ownership of this software and the [`nlohmann-json` package in PyPI](https://pypi.org/project/nlohmann-json/) to the [nlohmann](https://github.com/nlohmann) organization
  * If you would like any help to find a more clean way to repackage or push to [PyPI](https://pypi.org/project/nlohmann-json), we are very open to providing input.

**Comment by nlohmann** [comment]
Thanks for flagging the GitHub API rate-limit pain and for offering to transfer the PyPI repackaging over - broader, more reliable package availability benefits everyone pulling the library into automated builds.

Also worth noting: discussion #4844 hit exactly the gap this addresses -- a third-party `nlohmann-json` PyPI package missing CMake config files -- so linking the two here for anyone who lands on one and not the other.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
